rust-runtime/aws-smithy-http/src/byte_stream.rs

```diff
@@ -403,6 +427,29 @@ impl ByteStream {
     }
 }
 
+pin_project! {
+    // A new-type wrapper around `ByteStream` so we can pass it to `tokio_util::io::StreamReader`.
+    struct StreamWrapper {
+        #[pin]
+        byte_stream: ByteStream,
+    }
```

Comment:
This is pretty much the same as `HyperBodyWrapEventStream` in `hyper_body_wrap_stream.rs`. It'd be nice to remove duplication 🤔

Reply:
Moving `hyper_body_wrap_stream.rs` to a doc-hidden module in `aws-smithy-http` helped remove this one-off struct. See f5c8a96.